# Data Types

### Introduction

Ketika membuat variabel di Python, variabel tersebut memiliki nilai dengan tipe data yang sesuai. Ada banyak jenis tipe data, seperti integer, float, boolean, dan string, yang semuanya akan kita bahas dalam pelajaran ini. (Ini hanya merupakan sebagian kecil dari tipe data yang tersedia - ada juga dictionary, set, list, tuple, dan masih banyak lagi.)

Tipe data penting, karena mereka menentukan jenis tindakan apa yang dapat Anda lakukan dengan mereka. Misalnya, Anda dapat membagi dua nilai float, tetapi Anda tidak bisa membagi dua string. Misalnya, 12.0/2.0 masuk akal, tetapi "cat"/"dog" tidak.

Untuk menghindari kesalahan, kita perlu memastikan bahwa tindakan yang kita lakukan sesuai dengan tipe data yang kita miliki.

### Integers

Integer adalah jenis data dalam Python yang merepresentasikan bilangan bulat tanpa bagian pecahan, yang dapat berupa bilangan positif (1, 2, 3, ...), bilangan negatif (-1, -2, -3, ...), atau nol (0).

Pada sel kode di bawah ini, kita menetapkan sebuah variabel x dengan sebuah bilangan bulat. Selanjutnya, kita memverifikasi tipe data dari variabel tersebut menggunakan fungsi type(), yang hanya memerlukan nama variabel yang ingin kita periksa dimasukkan ke dalam tanda kurung. Ini membantu kita memastikan bahwa variabel x secara benar memiliki tipe data integer.

In [1]:
x = 14
print(x)
print(type(x))

14
<class 'int'>


Pada output di atas, <class 'int'> mengacu pada tipe data integer. Ini adalah representasi Python yang menunjukkan bahwa variabel tersebut adalah tipe data integer. Dalam konteks ini, 'int' singkatan dari 'integer'. Jadi, ketika Anda melihat <class 'int'>, itu berarti variabel tersebut adalah sebuah bilangan bulat dalam Python.

### Floats

Floats adalah tipe data dalam Python yang mewakili bilangan dengan bagian pecahan. Mereka dapat memiliki banyak angka setelah koma desimal. Misalnya, 3.14 adalah sebuah float karena memiliki bagian pecahan, sedangkan 5 adalah sebuah integer karena tidak memiliki bagian pecahan.

In [2]:
nearly_pi = 3.141592653589793238462643383279502884197169399375105820974944
print(nearly_pi)
print(type(nearly_pi))

3.141592653589793
<class 'float'>


Kita juga bisa menentukan sebuah float dengan pecahan. Misalnya, kita bisa menggunakan pecahan seperti 3/2 atau 1/3 untuk menentukan nilai float. Python akan menangani konversi dari pecahan ke float secara otomatis.

In [3]:
almost_pi = 22/7
print(almost_pi)
print(type(almost_pi))

3.142857142857143
<class 'float'>


Salah satu fungsi yang sangat berguna untuk pecahan adalah fungsi round(). Fungsi ini memungkinkan Anda untuk membulatkan sebuah angka menjadi sejumlah titik desimal tertentu yang ditentukan.

Misalnya, jika Anda memiliki angka float seperti 3.14159 dan Anda ingin membulatkannya menjadi dua angka desimal, Anda dapat menggunakan fungsi round() seperti ini: round(3.14159, 2). Hasilnya akan menjadi 3.14.

In [4]:
# Round to 5 decimal places
rounded_pi = round(almost_pi, 5)
print(rounded_pi)
print(type(rounded_pi))

3.14286
<class 'float'>


Ketika Anda menulis sebuah angka dengan titik desimal, Python akan mengenali angka tersebut sebagai tipe data float.

Misalnya, 1. (atau 1.0, 1.00, dan sebagainya) akan dikenali sebagai float. Hal ini terjadi, meskipun secara teknis angka-angka ini tidak memiliki bagian pecahan!

In [5]:
y_float = 1.
print(y_float)
print(type(y_float))

1.0
<class 'float'>


### Booleans

Booleans mewakili salah satu dari dua nilai: True atau False. Pada sel kode di bawah ini, z_one diatur sebagai boolean dengan nilai True.

In [6]:
z_one = True
print(z_one)
print(type(z_one))

True
<class 'bool'>


Berikut adalah kode untuk menetapkan variabel z_two dengan nilai boolean False

In [7]:
z_two = False
print(z_two)
print(type(z_two))

False
<class 'bool'>


Kode 1 < 2 adalah sebuah ekspresi perbandingan dalam Python yang menguji apakah angka 1 lebih kecil dari angka 2. Karena pernyataan ini benar dalam matematika, hasil ekspresi tersebut adalah True.

Selanjutnya, kita menyimpan hasil ekspresi ini ke dalam variabel z_three. Ini berarti kita mengaitkan nilai True ke dalam variabel z_three.

Jadi, ketika kita mencetak nilai z_three, kita akan mendapatkan output True, karena itu adalah nilai yang disimpan dalam variabel z_three.

In [8]:
z_three = (1 < 2)
print(z_three)
print(type(z_three))

True
<class 'bool'>


Dalam kode 5 < 3, kita mencoba untuk membandingkan apakah angka 5 lebih kecil dari angka 3. Namun, ini adalah pernyataan yang salah. Oleh karena itu, nilai yang disimpan dalam variabel z_four adalah False. Saat kita mencetak nilai z_four, kita akan melihat output False. Intinya, tipe data boolean (False) digunakan untuk merepresentasikan nilai kebenaran dari ekspresi matematika.

In [9]:
z_four = (5 < 3)
print(z_four)
print(type(z_four))

False
<class 'bool'>


Anda bisa mengubah nilai boolean dari True menjadi False, atau sebaliknya, dengan menggunakan operator not.

Contohnya, not True akan menghasilkan nilai False, dan not False akan menghasilkan nilai True.

In [10]:
z_five = not z_four
print(z_five)
print(type(z_five))

True
<class 'bool'>


### Strings

Tipe data string adalah kumpulan karakter (seperti huruf abjad, tanda baca, digit numerik, atau simbol) yang terdapat di dalam tanda kutip. String umumnya digunakan untuk merepresentasikan teks atau data berbasis teks dalam pemrograman.

String dapat dibuat menggunakan tanda kutip tunggal (' ') atau tanda kutip ganda (" "). Dalam Python, keduanya sama-sama valid dan dapat digunakan untuk membuat string. Misalnya, 'Hello' dan "Hello" akan menghasilkan string yang sama, yaitu "Hello".

In [11]:
w = "Hello, Python!"
print(w)
print(type(w))

Hello, Python!
<class 'str'>


Anda dapat mengambil panjang (jumlah karakter) dari sebuah string dengan menggunakan fungsi len(). Misalnya, string "Hello, Python!" memiliki panjang 14, karena terdiri dari 14 karakter, termasuk spasi, koma, dan tanda seru. Penting untuk dicatat bahwa tanda kutip (baik tunggal maupun ganda) tidak dihitung saat menghitung panjang string.

In [12]:
print(len(w))

14


Ada jenis string khusus yang disebut sebagai string kosong (empty string), yang memiliki panjang nol. Ini berarti bahwa tidak ada karakter yang ada di dalam string tersebut.

In [13]:
shortest_string = ""
print(type(shortest_string))
print(len(shortest_string))

<class 'str'>
0


Ketika Anda menempatkan sebuah angka di dalam tanda kutip, Python akan menganggapnya sebagai string, bukan sebagai nilai numerik.

In [14]:
my_number = "1.12321"
print(my_number)
print(type(my_number))

1.12321
<class 'str'>


Anda dapat menggunakan fungsi float() untuk mengonversi string menjadi nilai float jika string tersebut dapat dikonversi ke dalam bentuk float.

Misalnya, jika Anda memiliki string "10.43430" atau "3", keduanya dapat dengan mudah dikonversi menjadi nilai float menggunakan fungsi float():



In [15]:
also_my_number = float(my_number)
print(also_my_number)
print(type(also_my_number))

1.12321
<class 'float'>


 Sama halnya seperti Anda dapat menambahkan dua angka (float atau integer) menggunakan operator penambahan (+), Anda juga dapat menambahkan dua string menggunakan operator yang sama. Ini akan menghasilkan sebuah string yang lebih panjang yang menggabungkan dua string asli dengan cara menggabungkannya (concatenation).

In [16]:
new_string = "abc" + "def"
print(new_string)
print(type(new_string))

abcdef
<class 'str'>


Anda tidak dapat melakukan pengurangan atau pembagian dengan dua string, karena operasi tersebut tidak didefinisikan untuk tipe data string dalam Python.

Namun, Anda dapat mengalikan sebuah string dengan sebuah integer. Hal ini akan menghasilkan string yang sama, digabungkan dengan dirinya sendiri sejumlah kali yang ditentukan oleh nilai integer tersebut.

Misalnya, jika Anda memiliki string "abc" dan Anda mengalikannya dengan angka 3, Anda akan mendapatkan hasil "abcabcabc", yaitu string "abc" yang digabungkan dengan dirinya sendiri sebanyak tiga kali.

In [17]:
newest_string = "abc" * 3
print(newest_string)
print(type(newest_string))

abcabcabc
<class 'str'>


In [18]:
will_not_work = "abc" * 3.

TypeError: can't multiply sequence by non-int of type 'float'

Anda mencoba untuk mengalikan sebuah string dengan sebuah bilangan pecahan (float), yaitu 3.0. Namun, di Python, Anda hanya dapat mengalikan sebuah string dengan sebuah bilangan bulat (integer).

Ini karena Python tidak mendukung penggandaan string dengan bilangan pecahan. Jadi, Anda harus mengubah bilangan pecahan tersebut menjadi bilangan bulat terlebih dahulu sebelum mengalikan dengan string, jika itu adalah yang Anda inginkan. Misalnya, Anda bisa menggunakan int() untuk mengubah 3.0 menjadi 3, kemudian melakukan penggandaan string dengan bilangan bulat tersebut.

In [19]:
will_work = "abc" * 3
print(will_work)

abcabcabc


Dalam kode di atas, kita mengalikan string "abc" dengan bilangan bulat 3. Ini menghasilkan string "abc" yang digabungkan dengan dirinya sendiri sebanyak tiga kali, seperti yang diharapkan.